### PR TITLE
[decoupled-execution] Fix try_commit

### PR DESCRIPTION
**Pull request by yuxiamit**
The function try_commit should process batch by batch.

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update or suggest changes to the docs at https://developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: consensus/src/block_storage/block_store.rs [hunk 1]
```diff
@@ -166,15 +166,22 @@ impl BlockStore {
...
-        // B_i and figure out we only have B_j.
-        // Here we commit up to the highest_commit_cert to maintain highest_commit_cert == state_computer.committed_trees.
-        if self.highest_ordered_cert().commit_info().round() > self.ordered_root().round() {
-            let finality_proof = self.highest_ordered_cert().ledger_info().clone();
-            if let Err(e) = self.commit(finality_proof).await {
-                error!(error = ?e, "Commit error during build/rebuild");
+        // reproduce the same batches (important for the commit phase)
+
+        let mut certs = self.inner.read().get_all_quorum_certs_with_commit_info();
+        certs.sort_unstable_by_key(|qc| qc.commit_info().round());
```
Comment (zekun000) [inline consensus/src/block_storage/block_store.rs:172]: why need unstable one?

Comment (yuxiamit) [reply consensus/src/block_storage/block_store.rs:172]: this is a bit faster than the stable sort. We do not require the stability here.

Comment (yuxiamit) [reply consensus/src/block_storage/block_store.rs:172]: hmm, maybe if the rounds are different, there is no difference between the two?

Comment (zekun000) [reply consensus/src/block_storage/block_store.rs:172]: round should be unique here?

Comment (yuxiamit) [reply consensus/src/block_storage/block_store.rs:172]: yeah in this case the result will be the same. But I think rust stable and unstable sort are actually two different algorithms and the unstable one never allocate extra memory.

**Review by zekun000 — DISMISSED** [review]

**Pull request closed without merging**